pkg/authorization/api/validation/validation.go

```diff
@@ -8,48 +8,126 @@ import (
+	}
+
+	return allErrs
+}
+
+func ValidatePolicyBindingUpdate(oldPolicyBinding *authorizationapi.PolicyBinding, policyBinding *authorizationapi.PolicyBinding) fielderrors.ValidationErrorList {
+	allErrs := ValidatePolicyBinding(policyBinding)
+	allErrs = append(allErrs, validation.ValidateObjectMetaUpdate(&oldPolicyBinding.ObjectMeta, &policyBinding.ObjectMeta).Prefix("metadata")...)
+
+	if oldPolicyBinding.PolicyRef.Namespace != policyBinding.PolicyRef.Namespace {
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("policyRef.Namespace", policyBinding, "cannot change policyRef"))
+	}
+
 	return allErrs
 }
 
+func ValidateRoleName(name string, prefix bool) (bool, string) {
+	return true, ""
+}
+
 // ValidateRole tests required fields for a Role.
 func ValidateRole(role *authorizationapi.Role) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
+	allErrs = append(allErrs, validation.ValidateObjectMeta(&role.ObjectMeta, true, ValidateRoleName).Prefix("metadata")...)
 
-	if len(role.Name) == 0 {
-		allErrs = append(allErrs, fielderrors.NewFieldRequired("name"))
-	}
-
-	allErrs = append(allErrs, validation.ValidateLabels(role.Labels, "labels")...)
 	return allErrs
 }
 
+func ValidateRoleBindingName(name string, prefix bool) (bool, string) {
+	return true, ""
+}
+
 // ValidateRoleBinding tests required fields for a RoleBinding.
 func ValidateRoleBinding(roleBinding *authorizationapi.RoleBinding) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
-
-	if len(roleBinding.Name) == 0 {
-		allErrs = append(allErrs, fielderrors.NewFieldRequired("name"))
-	}
+	allErrs = append(allErrs, validation.ValidateObjectMeta(&roleBinding.ObjectMeta, true, ValidateRoleBindingName).Prefix("metadata")...)
 
 	if len(roleBinding.RoleRef.Namespace) == 0 {
 		allErrs = append(allErrs, fielderrors.NewFieldRequired("roleRef.namespace"))
 	} else if !util.IsDNS1123Subdomain(roleBinding.RoleRef.Namespace) {
-		allErrs = append(allErrs, fielderrors.NewFieldInvalid("roleRef.namespace", roleBinding.RoleRef.Namespace, "roleRef.namespace must be a valid subdomain"))
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("roleRef.namespace", roleBinding, "roleRef.namespace must be a valid subdomain"))
+	}
+
+	if len(roleBinding.RoleRef.Name) == 0 {
+		allErrs = append(allErrs, fielderrors.NewFieldRequired("roleRef.name"))
+	} else {
+		if valid, err := ValidateRoleName(roleBinding.RoleRef.Name, false); !valid {
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid("roleRef.name", roleBinding, err))
+		}
+	}
+
+	return allErrs
+}
+
+func ValidateRoleBindingUpdate(oldRoleBinding *authorizationapi.RoleBinding, roleBinding *authorizationapi.RoleBinding) fielderrors.ValidationErrorList {
+	allErrs := ValidateRoleBinding(roleBinding)
+	allErrs = append(allErrs, validation.ValidateObjectMetaUpdate(&oldRoleBinding.ObjectMeta, &roleBinding.ObjectMeta).Prefix("metadata")...)
```

Comment:
ValidateObjectMetaUpdate requires resource version... what are you deriving the resource version of a rolebinding from?